File: SantanderTeste/app/src/main/java/com/example/alessandrofsouza/santanderapp/presentation/pages/contact/ContactPresenter.java
package com.example.alessandrofsouza.santanderapp.presentation.pages.contact;

import android.support.annotation.NonNull;
import android.util.Log;

import com.example.alessandrofsouza.santanderapp.data.service.ApiService;
import com.example.alessandrofsouza.santanderapp.domain.model.ContactModel;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;


public class ContactPresenter implements ContactContract.Presenter{

    private static final String TAG = "Santander ";
    private ContactContract.View view;
    private ApiService service;

    public ContactPresenter(@NonNull ContactContract.View contractView) {
        view = contractView;
        view.setPresenter(this);
        service = new ApiService();
    }

    @Override
    public void init() {
        getContactCells();
    }

    public void getContactCells() {
        service.getApi().listCells().enqueue(new Callback<ContactModel>() {
            @NonNull
            @Override
            public void onResponse(Call<ContactModel> call, Response<ContactModel> response) {
                if (response.isSuccessful()) {
                    ContactModel contactModel = response.body();
                    view.showContactCells(contactModel.getCells());

                } else {
                    Log.e(TAG, "Error Unsuccessful " + response.errorBody());
                }
            }

            @Override
            public void onFailure(Call<ContactModel> call, Throwable t) {
                Log.e(TAG, "Error Failure " + t.getMessage());
            }
        });
    }


}

File: source/app/src/main/java/br/com/andreyneto/testesantander/ui/contact/ContactPresenter.kt
package br.com.andreyneto.testesantander.ui.contact

import br.com.andreyneto.testesantander.model.ContactResponse
import br.com.andreyneto.testesantander.service.ApiService
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class ContactPresenter(
        val view: ContactContract.View): ContactContract.Presenter {
    override fun sendForm() {
        view.formSended()
    }

    override fun newMessage() {
        view.showForm()
    }

    private val apiService = ApiService()

    init {
        view.setPresenter(this)
    }

    override fun getCells() {

        apiService.getApi().cells.enqueue(object : Callback<ContactResponse> {
            override fun onResponse(call: Call<ContactResponse>, response: Response<ContactResponse>) {
                if(response.isSuccessful) view.showCells(response.body()?.cells!!)
            }
            override fun onFailure(call: Call<ContactResponse>, t: Throwable) {

            }
        })
    }

    override fun start() {
        getCells()
    }
}